[PyTorch] Use DimVector in at::matmul

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #72230

Here's a small PR that only fixes the extra heap allocations for shapes. Hopefully won't get stuck like #64387.

Differential Revision: [D33962610](https://our.internmc.facebook.com/intern/diff/D33962610/)

swolchok committed:
[PyTorch] Use DimVector in at::matmul

Here's a small PR that only fixes the extra heap allocations for shapes. Hopefully won't get stuck like #64387.

Differential Revision: [D33962610](https://our.internmc.facebook.com/intern/diff/D33962610/)

[ghstack-poisoned]

Lezcano commented:
I should spend some time splitting that PR into more manageable chunks to be able to isolate that mobile failure...

IvanYashchuk reviewed (approved):
Maybe `tensor.sizes().vec()` should be disallowed? :)

swolchok committed:
Update on "[PyTorch] Use DimVector in at::matmul"


Here's a small PR that only fixes the extra heap allocations for shapes. Hopefully won't get stuck like #64387.

Differential Revision: [D33962610](https://our.internmc.facebook.com/intern/diff/D33962610/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Use DimVector in at::matmul"


Here's a small PR that only fixes the extra heap allocations for shapes. Hopefully won't get stuck like #64387.

Differential Revision: [D33962610](https://our.internmc.facebook.com/intern/diff/D33962610/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Use DimVector in at::matmul"


Here's a small PR that only fixes the extra heap allocations for shapes. Hopefully won't get stuck like #64387.

Differential Revision: [D33962610](https://our.internmc.facebook.com/intern/diff/D33962610/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Use DimVector in at::matmul"


Here's a small PR that only fixes the extra heap allocations for shapes. Hopefully won't get stuck like #64387.

Differential Revision: [D33962610](https://our.internmc.facebook.com/intern/diff/D33962610/)

[ghstack-poisoned]
